src/glyphs.js

```diff
@@ -142,17 +144,29 @@ const updateMessages = (id, path, req) => {
 /* Constant for what the name in the "gauge" field of the messages spreadsheet
 * should be when creating a message attached to a view.
 */
-const viewPath = "intro"
+const viewPath = "intro";
 
-const importMessages = (line) => {
+const importMessage = (line, metaTypes) => {
   const message = line.split(FIELD_SEPERATOR);
+  const viewMessage = message[1].localeCompare(viewPath, undefined, { sensitivity: 'base' }) === 0;
 
-  if (message.length !== 4 && message.length !== 8)
+  if ((metaTypes.length === 0 && message.length > METADATA_START_INDEX) || (metaTypes.length !== 0 && message.length < METADATA_START_INDEX+1))
     return ERROR_STRING;
 
-  const viewMessage = (message[1] === viewPath);
+  let metaText = [];
+  for (let i = METADATA_START_INDEX; i < message.length; ++i) {
+    if (message[i] !== '') metaText.push(message[i]);
+  }
+
+  let metaArray = {};
+  for (let i = 0; i < metaTypes.length; i++) {
+    if (metaText[i])
+      metaArray[metaTypes[i]] = metaText[i];
+  }
 
-  const newMessage = {"text": message[2], "probability": (viewMessage) ? Number(message[3]) : message.splice(3, 5).map(num => Number(num))};
+  let newMessage = {"text": message[2], "probability": (viewMessage) ? Number(message[3]):message.splice(3, 5).map(num => Number(num))};
+
+  if (metaTypes.length > 0 &&  metaText.length > 0) newMessage = {...newMessage, "metadata" : metaArray};
```

Comment:
Would using an `&&` instead of the `if` be clearer?

Reply:
`if` works better for my brain here - if I were to use `&&`, I like for it to be only on the right-hand side, like `newMessage = metaTypes.length > 0 && metaText.length > 0 && {...newMessage, "metadata": metaArray}`

Reply:
Oops - and that expression would change `newMessage` either way, which is not what we want.  I guess the chained `&&` could work, but then it would be using `&&` as flow and I usually just like to use it as yielding expressions.